generator/internal/service_generator.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "generator/internal/service_generator.h"
+#include "absl/memory/memory.h"
+#include "absl/strings/str_cat.h"
+#include "generator/internal/codegen_utils.h"
+#include "generator/internal/stub_generator.h"
+#include <google/api/client.pb.h>
+#include <google/protobuf/descriptor.h>
+
+namespace google {
+namespace cloud {
+namespace generator_internal {
+
+ServiceGenerator::ServiceGenerator(
+    google::protobuf::ServiceDescriptor const* service_descriptor,
+    google::protobuf::compiler::GeneratorContext* context,
+    std::map<std::string, std::string> command_line_vars)
+    : descriptor_(service_descriptor), vars_(std::move(command_line_vars)) {
+  SetVars();
+  class_generators_.push_back(
```

Comment:
Question: I assume `class_generators_` will eventually have more than one element. How will those other class generators be added to this vector?